accounts/abi/bind/base.go

```diff
@@ -117,11 +117,14 @@ func DeployContract(opts *TransactOpts, abi abi.ABI, bytecode []byte, backend Co
 // sets the output to result. The result type might be a single field for simple
 // returns, a slice of interfaces for anonymous returns and a struct for named
 // returns.
-func (c *BoundContract) Call(opts *CallOpts, result interface{}, method string, params ...interface{}) error {
+func (c *BoundContract) Call(opts *CallOpts, results *[]interface{}, method string, params ...interface{}) error {
 	// Don't crash on a lazy user
 	if opts == nil {
 		opts = new(CallOpts)
 	}
+	if results == nil {
```

Comment:
that doesn't seem necessary since you have another check for `len(results) == 0` before results is used for the first time.

Reply:
Unfortunately not, as I have to dereference result in the length check. `len(*results) == 0`, If results is nil the dereference panics. It is valid for the user to pass in a valid slice pointer with length 0 which we will fill with our result. If he's lazy, he might pass nil which means he doesn't need the result.